v3/tasks/20250214-v3-protocol-fee-controller-v2/test/task.fork.ts

```diff
@@ -0,0 +1,193 @@
+        admin.address
+      );
+
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(
+        await feeController.getActionId(feeController.interface.getSighash('setGlobalProtocolYieldFeePercentage')),
+        admin.address
+      );
+  });
+
+  it('replaces the old fee controller', async () => {
+    const vaultAsExtension = vaultExtension.attach(vault.address);
+    const oldFeeControllerAddress = await vaultAsExtension.getProtocolFeeController();
+
+    const vaultAsAdmin = vaultAdmin.attach(vault.address);
+    await vaultAsAdmin.connect(admin).setProtocolFeeController(feeController.address);
+
+    const newFeeControllerAddress = await vaultAsExtension.getProtocolFeeController();
+    expect(newFeeControllerAddress).not.to.eq(oldFeeControllerAddress);
+    expect(newFeeControllerAddress).to.eq(feeController.address);
+  });
+
+  it('sets non-zero global fees', async () => {
+    await feeController.connect(admin).setGlobalProtocolSwapFeePercentage(GLOBAL_SWAP_FEE_PERCENTAGE);
+    await feeController.connect(admin).setGlobalProtocolYieldFeePercentage(GLOBAL_YIELD_FEE_PERCENTAGE);
+
+    expect(await feeController.getGlobalProtocolSwapFeePercentage()).to.eq(GLOBAL_SWAP_FEE_PERCENTAGE);
+    expect(await feeController.getGlobalProtocolYieldFeePercentage()).to.eq(GLOBAL_YIELD_FEE_PERCENTAGE);
+  });
+
+  it('deploys a pool and gets the events', async () => {
+    const newWeightedPoolParams = {
+      name: 'Mock Weighted Pool',
+      symbol: 'TEST',
+      tokens: tokenConfig,
+      normalizedWeights: [fp(0.8), fp(0.2)],
+      roleAccounts: {
+        pauseManager: ZERO_ADDRESS,
+        swapFeeManager: ZERO_ADDRESS,
+        poolCreator: ZERO_ADDRESS,
+      },
+      swapFeePercentage: fp(0.01),
+      hooksAddress: ZERO_ADDRESS,
+      enableDonations: false,
+      disableUnbalancedLiquidity: false,
+      salt: ONES_BYTES32,
+    };
+
+    poolCreationReceipt = await (
+      await factory.create(
+        newWeightedPoolParams.name,
+        newWeightedPoolParams.symbol,
+        newWeightedPoolParams.tokens,
+        newWeightedPoolParams.normalizedWeights,
+        newWeightedPoolParams.roleAccounts,
+        newWeightedPoolParams.swapFeePercentage,
+        newWeightedPoolParams.hooksAddress,
+        newWeightedPoolParams.enableDonations,
+        newWeightedPoolParams.disableUnbalancedLiquidity,
+        newWeightedPoolParams.salt
+      )
+    ).wait();
+
+    const event = expectEvent.inReceipt(poolCreationReceipt, 'PoolCreated');
+    pool = await task.instanceAt(POOL_CONTRACT_NAME, event.args.pool);
+  });
+
+  it('checks pool deployment', async () => {
+    const poolTokens = (await pool.getTokens()).map((token: string) => token.toLowerCase());
+    expect(poolTokens).to.be.deep.eq(tokenConfig.map((config) => config.token.toLowerCase()));
+  });
+
+  it('pool creation emits initial fee events', async () => {
```

Comment:
How about checking that the protocol fee percentage for the pool is the new global fee percentage?